src/composables/useNetwork.spec.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  getSubdomainNetworkRedirectUrl,
+  getSubdomain,
+  handleNetworkUrl,
+} from './useNetwork';
+
+jest.mock('@/services/web3/useWeb3');
+
+describe('Navigation guards', () => {
+  it('should get correct top level domains', () => {
+    const urls = [
+      { path: 'polygon.balancer.fi/#/test', result: 'polygon' },
+      { path: 'app.balancer.fi', result: 'app' },
+      { path: 'balancer.fi', result: 'balancer' },
+      { path: 'beta.balancer.fi/trade', result: 'balancer' },
+      { path: 'beta.goerli.balancer.fi', result: 'goerli' },
+      { path: 'beta.balancer.fi/#/claim/test', result: 'balancer' },
+      { path: 'localhost:8080', result: 'localhost:8080' },
+      { path: 'localhost', result: 'localhost' },
+    ];
+
+    urls.forEach(url => {
+      expect(getSubdomain(url.path)).toEqual(url.result);
+    });
+  });
+
+  it('should get correct url redirects for old format urls', () => {
+    const urls = [
+      {
+        url: 'https://polygon.balancer.fi/#/pool/create',
+        result: 'https://localhost:8080/#/polygon/pool/create',
+      },
+      {
+        url: 'https://goerli.balancer.fi/#/pool/0x5a6a8cffb4347ff7fc484bf5f0f8a2e234d34255000200000000000000000275',
+        result:
+          'https://localhost:8080/#/goerli/pool/0x5a6a8cffb4347ff7fc484bf5f0f8a2e234d34255000200000000000000000275',
+      },
+      {
+        url: 'https://ethereum.balancer.fi/#/portfolio',
+        result: 'https://localhost:8080/#/ethereum/portfolio',
+      },
```

Comment:
The ethereum subdomain doesn't exist. Mainnet is on app.balancer.fi. I'd like to see a test case that insures `app.balancer.fi/#/some/path` basically doesn't redirect. Or redirects to `app.balancer.fi/#/ethereum/some/path` if it needs to.

I also noticed that `app.balancer.fi/#/trade`, for example, doesn't work. It doesn't redirect and it just loads the home page. That means any external link to out current mainnet trade page will be broken. We need to handle stuff like this.

Note: I've referenced `app.balancer.fi` as the domain here instead of the preview URL to keep it simple.